packages/wdio-browserstack-service/src/ai-handler.ts

```diff
@@ -0,0 +1,160 @@
+import aiSDK from '@browserstack/ai-sdk-node'
+import { BStackLogger } from './bstackLogger.js'
+import { TCG_URL, TCG_INFO, SUPPORTED_BROWSERS_FOR_AI } from './constants.js'
+import { handleHealingInstrumentation } from './instrumentation/funnelInstrumentation.js'
+import { createRequire } from 'node:module'
+
+import type { Capabilities } from '@wdio/types'
+import type BrowserStackConfig from './config.js'
+import type { Options } from '@wdio/types'
+import type { BrowserstackHealing } from '@browserstack/ai-sdk-node'
+import path from 'node:path'
+import fs from 'node:fs'
+import url from 'node:url'
+import { getBrowserStackUserAndKey } from './util.js'
+import type { BrowserstackOptions } from './types.js'
+
+class AiHandler {
+    authResult: BrowserstackHealing.InitSuccessResponse | BrowserstackHealing.InitErrorResponse
+    wdioBstackVersion: string
+    constructor() {
+        this.authResult = JSON.parse(process.env.TCG_AUTH_RESULT || '{}')
+        this.wdioBstackVersion = createRequire(import.meta.url)('../package.json').version
+    }
+
+    async authenticateUser(user: string, key: string) {
+        return await aiSDK.BrowserstackHealing.init(key, user, TCG_URL, this.wdioBstackVersion)
+    }
+
+    async updateCaps(
+        authResult: BrowserstackHealing.InitSuccessResponse | BrowserstackHealing.InitErrorResponse,
+        options: BrowserstackOptions,
```

Comment:
**Can we keep method signature same across fns, will produce less issues when users upgrade to higher versions?**

Reply:
Yes, this new type was added in the commit: https://github.com/xxshubhamxx/webdriverio_ai/pull/4/commits/e13c3d00ceac495b21bbb637dc3f877f204405fd to change reading of the flags from within the service level config. This change needs to be ported to v9 and v7.